Khan/perseus
# Add per-package coverage reporting [work-in-progress]

## Summary:
Flow had a command to generate coverage reports for a whole project, but TS doesn't have this capability.  There is a third party package called typescript-coverage-report that provides this capability.  I added a npm script "type:coverage" that can be used to produce type coverage reports for an individual package.  For some reason it doesn't work with the base tsconfig.json.

Issue: None

TODO:
- [ ] try to figure out the issue with the root tsconfig.json
- [ ] experiment with the "historyFile" setting described in https://github.com/plantain-00/type-coverage#config-in-packagejson

## Test plan:
- `$PKG=perseus yarn type:coverage`
- see that it prints the report in the terminal
- open ts-coverage/index.html to see an HTML version of the repo

<img width="1904" alt="Screen Shot 2023-04-12 at 10 45 58 AM" src="https://user-images.githubusercontent.com/1044413/231494790-f92b083d-53fd-4acd-a049-b24aa83d8c93.png">
<img width="1904" alt="Screen Shot 2023-04-12 at 10 46 14 AM" src="https://user-images.githubusercontent.com/1044413/231494807-3861e7d0-662e-4a01-a4cb-5c5282deb2e3.png">
